OpenLaTeXStudio/EditorModule/src/latexstudio/editor/menu/GeneratePDF.java

```diff
@@ -44,18 +45,44 @@
     @Override
     public void actionPerformed(ActionEvent e) {        
         File file = FileChooserService.getSelectedFile("pdf", "PDF files", DialogType.PDF, false);
+        int reply = JOptionPane.NO_OPTION;
         if (file != null) {
-            String filename = file.getName();
+            //When the user clicks a file, the Chooser shows "file.pdf", so we have to remove this ".pdf"
+            //Otherwise, pdflatex will create a "file.pdf.pdf"
+            if (file.toString().endsWith(".pdf")) {
+               file = new File(file.toString().substring(0, file.toString().length() - 4));
+            }
             
-            LOGGER.log("Invoking pdflatex");
-  
-            CommandLineExecutor.executeGeneratePDF(new CommandLineBuilder()
-                .withPathToSource(ApplicationUtils.getTempSourceFile())
-                .withOutputDirectory(file.getParentFile().getAbsolutePath())
-                .withJobname(filename)
-                .withWorkingFile(etc.getCurrentFile())
-                .withLatexPath(etc.getLatexPath())
-                .withLogger(new ApplicationLogger("pdflatex")));
+            //If the user only type "file", we have to check if "file.pdf" exists.
+            File v_overwriting = new File(file.toString() + ".pdf");
+            while (v_overwriting.exists() && reply == JOptionPane.NO_OPTION) {
+                reply = JOptionPane.showConfirmDialog(null,
+                            "The selected file already exists. Do you want to overwrite it?",

```

Comment:
Can we replace "selected file" with the full path to that file? This would give user additional confirmation and reminder of what was selected.